Sources/Build/BuildPlan/BuildPlan.swift

```diff
@@ -288,7 +288,7 @@ public class BuildPlan: SPMBuildCore.BuildPlan {
         self.fileSystem = fileSystem
         self.observabilityScope = observabilityScope.makeChildScope(description: "Build Plan")
 
-        var productMap: [ResolvedProduct.ID: (product: ResolvedProduct, buildDescription: ProductBuildDescription)] = [:]
+        var productMap = [ResolvedProduct.ID: (product: ResolvedProduct, buildDescription: ProductBuildDescription)]()
```

Comment:
I don't particularly like this change. Simply typing the value and using the empty ctor seems better.